### DISCUSSION TITLE
Roadmap updated

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#1579] [General]
Today I bring you news that may not be entirely unexpected: we have updated [the roadmap](https://github.com/twinbasic/twinbasic/issues/335).

All planned work has been extended by six months, in order to accommodate the efforts for our v1 release. Each day brings us one step closer to our goal, and this extra period will allow us to tie up all the loose ends and apply that final layer of polish.

Although the delay is frustrating, I hope you can all recognize the substantial strides we're making. For example, we've just implemented support for control arrays (including runtime loading/unloading of controls), and we're on the cusp of delivering MDI form support. These are significant achievements that will aid the migration of more VB6 projects to twinBASIC.

Thank you for your understanding and patience, and I look forward to delivering a well-rounded and high-quality v1 release in the not too distant future.

**Comment by gregse99** [comment]
You’re doing God’s work Wayne.  Keep up the great work.

**Comment by fafalone** [comment]
It's a long delay but probably for the best; better to deliver a complete, solid product even if that takes extra time (though the AAA game studios would seem to disagree 😄)  Most people understand the magnitude of the task, and can see tB marching towards it's goal by the day. No doubt the wait will be worth it.

**Comment by cabaptista** [comment]
Don´t give up. I heared about your software 3 months ago and i think you're doing an excelent job. And to prove it, i already have a simple browser written in twinBASIC. :)

Right now, i'm writing this message from it. Check screenshots.
  
![Image#1](https://github.com/twinbasic/twinbasic/assets/112648096/15e47027-a092-4387-a9b5-c55a108e2d3a)
![Image#2](https://github.com/twinbasic/twinbasic/assets/112648096/1f5882ab-dadf-43e8-b28d-b42324e8b1b5)

**Reply by WaynePhillipsEA** [reply]
Nice work @cabaptista !  And thanks for your encouragement :)

**Comment by DavidHollman** [comment]
A slightly longer wait until tB is released is nothing compared to the length of time we've needed something like tB. Take your time to get it "right" and it will be worth it.  I periodically download a beta and try things out, the progress is impressive: tangible, consistent, and well thought out. Its obviously only a matter of time until I can start to use tB to rebuild my living VB6 projects and gain so many benefits. :)
